documentation/specs/src/economics/proof-of-stake/cubic-slashing.md

```diff
@@ -106,22 +102,25 @@ As a function, it can be drawn as (assuming $r_{\text{min}} = 1\%$):
 
-## Slashes
+When a misbehavior is detected, the validator is immediately *frozen*: no unbonds from this validator are allowed. The validator is also *jailed* by removing it from the validator set indefinitely, starting at the next epoch following the discovery. We do not change the validator set in the current epoch, and this is the only instance wherein the protocol changes the validator set without waiting for the pipeline offset. Ultimately, a validator is *unfrozen* automatically after its last known slash has been processed, but a transaction is required to *unjail* a validator and reinstate it into the validator set. 
 
-Slashes should lead to punishment for delegators who were contributing voting power to the validator at the height of the infraction, _as if_ the delegations were iterated over and slashed individually.
+We walk through the overall procedure in some more detail here. When a misbehavior is detected:
+1. Compute the infraction epoch from its height. If the infraction epoch is earlier than `unbonding_len` epochs before the current epoch, then do not accept it.
+2. Enqueue the slash to be processed in the epoch `infraction_epoch + unbonding_len + window_width + 1`. The `unbonding_len` is used to prevent stake used to misbehave from being fully unbonded and withdrawn. The `window_width` is needed to ensure that all slashes that would contribute to the cubic slash rate can be considered, and the additional `1` is needed for processing at the very beginning of an epoch.
+3. Jail (and freeze) the misbehaving validator and remove it from the validator set, effective at the beginning of the next epoch. Bonds can still be made to validators while frozen. Unbonds can be made from validators while they are jailed, but not while frozen. At the pipeline offset only, if the misbehaving validator was removed from the consensus set, replace it with the appropriate below-capacity validator.
 
-This can be implemented as a negative inflation rate for a particular block.
+Now, at the beginning of each new epoch, if there are slashes enqueued for this epoch, then:
+1. Compute the cubic slash rate for slashes in this epoch.
+For each validator that has enqueued slashes,
+2. Compute the rate for each of the validator slashes. The total rate by which this validator is slashed is the sum of these rates, capped at `1.0`. Slash the validator voting powers with the total rate, enforcing that the slashed voting power be 0 if it would otherwise become negative.
```

Comment:
We should link to the part of the spec that explains how slashes are lazily processed (still negative inflation, right?)

Reply:
Which part in particular are you talking about? There was previously one sentence about "negative inflation" at the bottom of this file, but that's it, so I removed it.

Reply:
In the `bonding-mechanism.md`. I'll make some updates there and link to the proper place